### PR TITLE
Suspend register in ZStream.effectAsyncInterrupt and harden test

### LINKED ISSUE #1206: Flaky stream test
```
[info]   Stream.effectAsyncInterrupt
[error]     x effectAsyncInterrupt Left
[error]  TIMEOUT: 60 seconds (TestRuntime.scala:18)
```

### PR DESCRIPTION
**Pull request by iravid**
Resolves #1206

### CODE REVIEW

**Comment by dariusrobson** [comment]
I see. Was the not wrapping in the UIO(..) the problem here or was it the test?

**Comment by iravid** [comment]
Just the test :-) Wrapped the registration in `UIO` just as a best practice.

**Comment by dariusrobson** [comment]
Ah Cool. Thanks for patching!

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 1]
```diff
@@ -1536,15 +1536,17 @@ object ZStream extends ZStreamPlatformSpecific {
...
-            eitherStream = register(
-              k =>
-                runtime.unsafeRunAsync_(
-                  k.foldCauseM(
-                    cause => output.offer(Take.Fail(cause)).unit,
-                    b => output.offer(Take.Value(b)).unit
-                  )
-                )
-            )
+            eitherStream <- UIO {
```
Comment (jdegoes) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]: Ouch.

**Pull request merged**